pkg/maps/nodemap/node_map_v2.go

```diff
@@ -0,0 +1,125 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright Authors of Cilium
+
+package nodemap
+
+import (
+	"fmt"
+	"net"
+	"unsafe"
+
+	"golang.org/x/sys/unix"
+
+	"github.com/cilium/cilium/pkg/ebpf"
+)
+
+// compile time check of MapV2 interface
+var _ MapV2 = (*nodeMapV2)(nil)
+
+const (
+	MapNameV2 = "cilium_node_map_v2"
+)
+
+// Map provides access to the eBPF map node.
+type MapV2 interface {
```

Comment:
Call me crazy, but if we named the existing Go types for the node ID map `Old` instead of naming the new types `V2`, wouldn't that significantly reduce how many places you need to touch all over the codebase? It would also mean we wouldn't be dragging a `V2` around for the next 10 releases :sweat_smile:

Reply:
Oh, this is a good idea, +1!

Reply:
My opinion here is to be very explicit about a new version of a map being introduced. 

In my mind, it makes things very explicit that the map has transitioned and can help with some meta-debugging, knowing that map is new. 

I don't quite see whats wrong with keeping the V2 naming around, indefinitely? There is no reason to "hide" that this map went through more then one revision?

Reply:
The issue isn't so much `V2` staying around; that's just the cherry on the cake if we can get rid of it. The issue IMO is the potential for backport hell if we ever need to backport this anywhere.

I understand the desire to be explicit about the new version. I'm not suggesting changing the map name on the filesystem; that can remain `v2` if we prefer. I'm only talking about the Golang side of things where the `V2` suffix change is a little bit "invasive".

Reply:
I'm not sure how much of the code doesn't change, since we need to update all the areas where MapV1 is used since the fundamental way we use the mapValue now changes. Prior to this, we just used uint16 almost everywhere as the map Value. 

I can give it a try tho, and see if it produces much less of a diff while not making the code confusing.

Reply:
@pchaigno do you feel strongly enough about this to block the PR. 

I was working thru this change and I ran into a few things that don't sit well with me.
1. Going back and renaming the existing references feels a bit odd, instead of leaving the existing implementation and introducing v2. 
2. When we begin to try to rename the old map and use the existing map name, we destroy the commit structure I was after, where one commit cleanly introduces the new map, and another does the cut over. This needs to be smashed into one large commit if we want the commit to build, not quite a fan of this. 
3. I strongly believe any issues of repetitive rebasing can be fixed by enable `git rerere`. This solves most of the annoying rebase issues I run into when backporting.

Reply:
To be clear, "it sounds like a good idea" is not blocking feedback from me, so it's up to @pchaigno.

Reply:
We discussed this offline and concluded that conflicts are unlikely outside of the normal backporting path to v1.15, which Louis will likely handle.